packages/daemon/src/app/controller/base.ts

```diff
@@ -0,0 +1,68 @@
+import { Context, inject } from 'midway';
+import { ObjectSchema } from 'joi';
+import * as HttpStatus from 'http-status';
+import ServerSentEmitter from '../../utils/emitter';
+import { LogManager } from '../../service/log-manager';
+
+export class BaseController {
+  @inject()
+  ctx: Context;
+  /**
+   * validata the data by schema
+   * @param schema the schema
+   * @param data the data validated
+   */
+  validate(schema: ObjectSchema, data: any) {
+    const { error } = schema.validate(data);
+    if (error) {
+      this.ctx.throw(error.message, HttpStatus.BAD_REQUEST);
+    }
+  }
+  /**
+   * send successful response, the status will be set to 204 or 200,
+   * depend on if data is undefined
+   * @param data data returned
+   * @param status success code, should be [200, 299], default is 200
+   */
+  success(data?: any, status?: number): void {
```

Comment:
Is it necessary to judge whether the data exist or not first？

Reply:
I think we need to check the type of `data`, if `data` equals `0`, the status will be `HttpStatus.NO_CONTENT`, it's not correct.
``` js
this.ctx.status = data ? HttpStatus.OK : HttpStatus.NO_CONTENT;
```